src/test/java/FlightsReporterSpockTests.groovy

```diff
@@ -0,0 +1,197 @@
+        new Flights([shortFlight, shortFlight2, mediumLenFlight, longFlight]) | 300      | [longFlight]
+    }
+
+    @Unroll
+    def "get all #flights with #duration"() {
+        expect:
+        FlightsReporter.getAllFlightWithDuration(flights, duration).flights == expected
+
+        where:
+        flights                                                               | duration | expected
+        new Flights([])                                                       | 0        | []
+        new Flights([shortFlight, shortFlight2])                              | 0        | []
+        new Flights([shortFlight, shortFlight2])                              | 60       | [shortFlight, shortFlight2]
+        new Flights([shortFlight, shortFlight2, mediumLenFlight])             | 60       | [shortFlight, shortFlight2]
+        new Flights([shortFlight, shortFlight2, mediumLenFlight, longFlight]) | 120      | [mediumLenFlight]
+    }
+}
+
+class FlightsReporterPassengerTest extends Specification {
+    @Shared
+    Flight empty = FlightGenerator.generateFlight("passengers": 0)
+    @Shared
+    Flight small = FlightGenerator.generateFlight(["passengers": 50])
+    @Shared
+    Flight medium = FlightGenerator.generateFlight(["passengers": 150])
+    @Shared
+    Flight big = FlightGenerator.generateFlight(["passengers": 250])
+
+    def "get flight with most passengers"() {
+        expect:
+        FlightsReporter.getFlightWithMostPassengers(flights) == expected
+
+        where:
+        flights                                  | expected
+        new Flights([])                          | null
+        new Flights([empty])                     | empty
+        new Flights([empty, small, medium, big]) | big
+    }
+
+    def "get flight with least passengers"() {
+        expect:
+        FlightsReporter.getFlightWithLeastPassengers(flights) == expected
+
+        where:
+        flights                                  | expected
+        new Flights([])                          | null
+        new Flights([medium, big])               | medium
+        new Flights([empty, small, medium, big]) | empty
+    }
+
+    def "get #flights with passengers above #count"() {
+        expect:
+        FlightsReporter.getAllFlightsWithPassengersAbove(flights, count).flights == expected
+
+        where:
+        flights                                  | count | expected
+        new Flights([])                          | 0     | []
+        new Flights([empty])                     | -1    | [empty]
+        new Flights([empty])                     | 0     | [empty]
+        new Flights([empty])                     | 1     | []
+        new Flights([empty, small, medium, big]) | 100   | [medium, big]
+    }
+
+    def "get #flights with passengers below #count"() {
+        expect:
+        FlightsReporter.getAllFlightsWithPassengersBelow(flights, count).flights == expected
```

Comment:
NEW TASK: 

Flights - should send email via EmailService only if more than 2 flights are filtered.
- unit tests with mocks

**FlightsReporter** should send email via EmailService only if more than 2 flights are filtered.
- unit tests with mocks

Compare mocking for static methods vs not

Reply:
Try the above when EmailUtil is static